Src/Metrics.Tests/Reporting/EventMetricsCleanerTest.cs

```diff
@@ -0,0 +1,212 @@
+            var registry = new DefaultMetricsRegistry();
+            registry.Event("test", () => { return new EventMetric(); }, MetricTags.None);
+            var reportIndex = EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+
+            EventMetricsCleaner.UpdateTotalReportedEvents(reportIndex, registry.DataProvider.Events);
+
+            EventMetricsCleaner.GetReportsEventCount(reportIndex).Should().Be(1);
+        }
+
+        [Fact]
+        public void RemoveEvent_RemovesEventFromAllReports()
+        {
+            EventMetricsCleaner.Clear();
+            var registry = new DefaultMetricsRegistry();
+            registry.Event("test", () => { return new EventMetric(); }, MetricTags.None);
+            var reportIndex1 = EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+            var reportIndex2 = EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+            EventMetricsCleaner.UpdateTotalReportedEvents(reportIndex1, registry.DataProvider.Events);
+            EventMetricsCleaner.UpdateTotalReportedEvents(reportIndex2, registry.DataProvider.Events);
+
+            EventMetricsCleaner.RemoveEvent("test");
+
+            EventMetricsCleaner.GetReportsEventCount(reportIndex1).Should().Be(0);
+            EventMetricsCleaner.GetReportsEventCount(reportIndex2).Should().Be(0);
+        }
+
+        [Fact]
+        public void Clear_RemovesAllReports()
+        {
+            EventMetricsCleaner.Clear();
+            EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+
+            EventMetricsCleaner.Clear();
+
+            EventMetricsCleaner.TotalReports.Should().Be(0);
+        }
+
+        [Fact]
+        public void Clean_OnlyRemovesEventDetailItems_ThatHaveBeenReportedByAllReports()
+        {
+            EventMetricsCleaner.Clear();
+            var registry = new DefaultMetricsRegistry();
+            EventMetricsCleaner.Registry = registry;
+            var timer = new MockTimer();
+            EventMetricsCleaner.EnableTestTimer(timer);
+
+            var reportIndex1 = EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+            var reportIndex2 = EventMetricsCleaner.RegisterReport(new TimeSpan(0, 0, 0, 60));
+            var metric = new EventMetric();
+            registry.Event("test", () => { return metric; }, MetricTags.None);
+
+            metric.Record();
+            EventMetricsCleaner.UpdateTotalReportedEvents(reportIndex1, registry.DataProvider.Events);
+            metric.Record();
+            EventMetricsCleaner.UpdateTotalReportedEvents(reportIndex2, registry.DataProvider.Events);
+
+            EventMetricsCleaner.GetReportsReportedEventDetailCount(reportIndex1, "test").Should().Be(1);
+            EventMetricsCleaner.GetReportsReportedEventDetailCount(reportIndex2, "test").Should().Be(2);
+
+            timer.OnTimerCallback();
+
+            EventMetricsCleaner.GetEventDetailCount("test").Should().Be(1);
```

Comment:
Should this not rather look at the event list in the DefaultMetricsRegistry to see if it actually clears the event